packages/gatsby/src/bootstrap/__tests__/requires-writer.js

```diff
@@ -148,6 +149,10 @@ describe(`requires-writer`, () => {
 
     it(`should have index pages with higher priority than matchPaths`, async () => {
       const pages = generatePagesState([
+        {
+          path: `/another-custom-404`,
+          matchPath: `/*`,
+        },
```

Comment:
(to make sure order of created pages doesn't impact final order of sorted pages)